cmd/portal/internal/create_default_domain.go

```diff
@@ -0,0 +1,146 @@
+package internal
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/lib/pq"
+
+	corerand "github.com/authgear/authgear-server/pkg/util/rand"
+	"github.com/authgear/authgear-server/pkg/util/uuid"
+)
+
+type CreateDefaultDomainOptions struct {
+	DatabaseURL         string
+	DatabaseSchema      string
+	DefaultDomainSuffix string
+}
+
+func CreateDefaultDomain(opts CreateDefaultDomainOptions) (err error) {
+	db := openDB(opts.DatabaseURL, opts.DatabaseSchema)
+	ctx := context.Background()
+
+	tx, err := db.BeginTx(ctx, nil)
+	defer func() {
+		if err != nil {
+			err = tx.Rollback()
+		}
+	}()
+
+	allConfigSourceList, err := selectConfigSources(ctx, tx, nil)
+	if err != nil {
+		return
+	}
+
+	for _, configSource := range allConfigSourceList {
+		appID := configSource.AppID
+		domain := makeDefaultDomain(appID, opts.DefaultDomainSuffix)
+
+		var exists bool
+		exists, err = checkDomainExistance(ctx, tx, appID, domain)
+		if err != nil {
+			return
+		}
+
+		if !exists {
+			err = createDefaultDomain(ctx, tx, appID, domain)
+			if err != nil {
+				return
+			}
+			fmt.Printf("created: %s\n", domain)
+		} else {
+			fmt.Printf("skipped: %s\n", domain)
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func makeVerificationNonce() string {
+	nonce := make([]byte, 16)
+	corerand.SecureRand.Read(nonce)
+	verificationNonce := hex.EncodeToString(nonce)
+	return verificationNonce
+}
```

Comment:
Looks like this code was duplicated from `pkg/portal/service/domain.go`. Maybe we should create a util function for this to ensure their logics are same?